telemetry/src/lib.rs

```diff
@@ -68,24 +195,44 @@ impl TelemetryConfig {
         } else {
             None
         };
+
+        // If journald is not available or we're in a TTY, fallback to stderr
         let stderr_layer = journald_layer
             .is_none()
             .then(|| tracing_subscriber::fmt::layer().with_writer(std::io::stderr));
-        assert!(stderr_layer.is_some() || journald_layer.is_some());
+
+        // 3. If OTLP tracing is available, attach a tracing-opentelemetry layer
+        let otlp_layer = self.tracer_provider.as_ref().map(|provider| {
+            let tracer = provider.tracer("orb-telemetry");
+            tracing_opentelemetry::layer().with_tracer(tracer)
+        });
+
+        // 4. Build the final subscriber
         registry
             .with(tokio_console_layer)
             .with(stderr_layer)
             .with(journald_layer)
+            .with(otlp_layer)
             .with(self.global_filter)
             .try_init()
     }
 
-    /// Initializes the telemetry config. Call this only once, at the beginning of the
-    /// program.
-    ///
-    /// Calling this more than once or when another tracing subscriber is registered
-    /// will cause a panic.
+    /// Initializes telemetry, panicking if something goes wrong.
     pub fn init(self) {
-        self.try_init().expect("failed to initialize orb-telemetry")
+        self.try_init().expect("failed to initialize orb-telemetry");
+    }
+
+    /// Optional shutdown hook to flush any pending OTLP spans.
+    /// For journald/stderr, it's usually not necessary.
+    pub fn shutdown_tracing(&self) {
```

Comment:
I think the right way to do this is to instead have the `init` function return a struct called `TelemetryShutdownHandler`. On `Drop` we will call `global::shutdown_tracer_provider()`. We should then have a `#[must_use]` attribute on the type too, to ensure that people actually hang onto it.

This also allows us to remove the tracer provider from the config struct. This keeps the config struct a "builder", it isn't designed to hold state after init.